### ISSUE TITLE
bug: Possible bug in JavaVM::getEnv(..).

### ISSUE THREAD

**Issue by user-L9b40P** [SuperIceCN/zig-jni#1]
I've just started with Zig, and was very interested in this library! 😄 

I have the following code:

```zig
pub fn JNI_OnLoad(vm: *jni.JavaVM, _: ?*anyopaque) callconv(.C) jni.jint {
    // JNIEnv pointer.
    var env: jni.JNIEnv = undefined;

    // Call getEnv to retrieve the JNIEnv pointer, and destroy the VM if not succsesfull.
    vm.getEnv(&ptr, @intFromEnum(jni.JNIVersion.JNI1_6)) catch blk: {
        break :blk vm.destroy();
    };

    // Return the JNI version that is supported.
    return @intFromEnum(jni.JNIVersion.JNI1_6);
}
```

As I would like to get the JNIEnv from the VM when the native is initially loaded.

However I get this error on compilation:

```css
...\src\main\zig\lib.zig:198:75: error: expected type '[*c]?*anyopaque', found '**[*c]const cimport.struct_JNINativeInterface_'

return try checkError(self._cJavaVM.?.*.*.GetEnv.?(self._cJavaVM, &(penv._cJNIEnv), version));
                                                                  ^~~~~~~~~~~~~~~~
```

Which originates from within this library. Is this an error on my part, or the libraries? Sorry if it is my fault. 😅

**Comment by SuperIceCN** [comment]
Sorry for the late response. Could you please try replacing the line `lib.zig:198` with the following in your local cache and recompile it? This may fix the bug and if it works I'll release a new version. Thank you!
```zig
return try checkError(self._cJavaVM.?.*.*.GetEnv.?(self._cJavaVM, @ptrCast(penv), version));
```

**Comment by user-L9b40P** [comment]
Yep that works! This is what it looks like:

```zig
pub fn getEnv(self: *JavaVM, penv: *JNIEnv, version: jint) !void {
    return try checkError(self._cJavaVM.?.*.*.GetEnv.?(self._cJavaVM, @ptrCast(penv), version));
}
```

And for future reference if anyone finds this, my code above is very flawed, here are some corrections:

```zig
pub fn JNI_OnLoad(vm: *jni.cjni.JavaVM, _: ?*anyopaque) callconv(.C) jni.jint {
    // JavaVM pointer.
    const wrapped: *jni.JavaVM = @constCast(&jni.JavaVM{
        ._cJavaVM = vm,
    });
    
    // JNIEnv pointer.
    var env: jni.JNIEnv = undefined;

    // Call getEnv to retrieve the JNIEnv pointer.
    _ = wrapped.getEnv(&env, @intFromEnum(jni.JNIVersion.JNI1_6)) catch blk: {
        break :blk undefined;
    };

    // Return the JNI version that is supported.
    return @intFromEnum(jni.JNIVersion.JNI1_6);
}
```

- JNI_OnLoad takes the `cjni` type.
- You have to create a wrapper with that.
- Fixed the error handling.

**Comment by user-L9b40P** [comment]
On another note, do you think it's worth it to add a `warp` function like `JNIEnv` has to prevent needing to construct the `JavaVM` instance explicitly when using `cjni`?

**Comment by SuperIceCN** [comment]
> On another note, do you think it's worth it to add a `warp` function like `JNIEnv` has to prevent needing to construct the `JavaVM` instance explicitly when using `cjni`?

Yes, of course. I've released a new version `v0.0.9` which includes all the modification. If there are still problems, please feel free to contact me.
